pkg/cstorpool/v1alpha2/kubernetes_test.go

```diff
@@ -0,0 +1,422 @@
+}
+
+func fakeGetOk(cs *clientset.Clientset, name string,
+	opts metav1.GetOptions) (*apis.CStorPool, error) {
+	return &apis.CStorPool{}, nil
+}
+
+func fakeGetErr(cs *clientset.Clientset, name string,
+	opts metav1.GetOptions) (*apis.CStorPool, error) {
+	return &apis.CStorPool{}, errors.New("some error")
+}
+
+func fakeCreateOk(cs *clientset.Clientset,
+	obj *apis.CStorPool) (*apis.CStorPool, error) {
+	return &apis.CStorPool{}, nil
+}
+
+func fakeCreateErr(cs *clientset.Clientset,
+	obj *apis.CStorPool) (*apis.CStorPool, error) {
+	return &apis.CStorPool{}, errors.New("some error")
+}
+
+func fakePatchOk(cs *clientset.Clientset, name string,
+	pt types.PatchType, patchObj []byte) (*apis.CStorPool, error) {
+	return &apis.CStorPool{}, nil
+}
+
+func fakePatchErr(cs *clientset.Clientset, name string, pt types.PatchType,
+	patchObj []byte) (*apis.CStorPool, error) {
+	return &apis.CStorPool{}, errors.New("some error")
+}
+
+func fakeDeleteOk(cs *clientset.Clientset, name string,
+	opt *metav1.DeleteOptions) error {
+	return nil
+}
+
+func fakeDeleteErr(cs *clientset.Clientset, name string,
+	opt *metav1.DeleteOptions) error {
+	return errors.New("some error")
+}
+
+func fakeGetClientsetOk() (cs *clientset.Clientset, err error) {
+	return &clientset.Clientset{}, nil
+}
+
+func fakeGetClientsetNil() (clientset *clientset.Clientset, err error) {
+	return nil, nil
+}
+
+func fakeGetClientsetErr() (clientset *clientset.Clientset, err error) {
+	return nil, errors.New("Some error")
+}
+
+func TestWithDefaults(t *testing.T) {
+	tests := map[string]struct {
+		listFn             listFunc
+		getFn              getFunc
+		getClientsetFn     getClientsetFunc
+		createFn           createFunc
+		patchFn            patchFunc
+		deleteFn           delFn
+		expectList         bool
+		expectGet          bool
+		expectGetClientset bool
+		expectCreate       bool
+		expectPatch        bool
+		expectDelete       bool
+	}{
+		"When mockclient is empty": {nil, nil, nil, nil, nil, nil, true, true, true, true, true, true},
+		"When mockclient contains all of them": {fakeListOk, fakeGetOk,
+			fakeGetClientsetOk, fakeCreateOk, fakePatchOk, fakeDeleteOk, true, true, true, true, true, true},
+	}
+
+	for name, mock := range tests {
+		name := name //pin it
+		mock := mock // pin it
+		t.Run(name, func(t *testing.T) {
+			fc := &Kubeclient{}
+			fc.list = mock.listFn
+			fc.get = mock.getFn
+			fc.getClientset = mock.getClientsetFn
+			fc.create = mock.createFn
+			fc.patch = mock.patchFn
+
+			fc.withDefaults()
+
+			if (fc.list != nil) != mock.expectList {
+				t.Fatalf("test %s failed: expected non-nil fc.list but got %v", name, fc.list)
+			}
+
+			if (fc.get != nil) != mock.expectGet {
+				t.Fatalf("test %s failed: expected non-nil fc.get but got %v", name, fc.get)
+			}
+
+			if (fc.getClientset != nil) != mock.expectGetClientset {
+				t.Fatalf("test %s failed: expected non-nil fc.getClientset but got %v", name, fc.getClientset)
```

Comment:
How can we be sure of this message?
expect can be true or false.

Reply:
will do this in a refactor PR.